### ISSUE TITLE
Case sensitive UUIDs

### ISSUE THREAD

**Issue by nicklars3n** [Polidea/blemulator_flutter#111]
When creating a `SimulatedService` or `SimulatedCharacteristic` with an upper case uuid, they are not found when using the uuids returned during discovery.

Example:
Using the SensorTag `SimulatedPeripheral` with upper case uuids
```dart
await peripheral.discoverAllServicesAndCharacteristics();

List<Service> services = await peripheral.services();

for (Service service in services) {
    print('Service: ${service.uuid}');

    List<Characteristic> characteristics = await peripheral.characteristics(service.uuid);

    for (Characteristic characteristic in characteristics) {
        print('  Characteristic: ${characteristic.uuid}');
    }
}
```
Prints only
```
Service: f000aa00-0451-4000-b000-000000000000
```

Using the workaround `service.uuid.toUpperCase()`
```dart
List<Characteristic> characteristics = await peripheral.characteristics(service.uuid.toUpperCase());
```
Prints:
```
Service: f000aa00-0451-4000-b000-000000000000
  Characteristic: f000aa01-0451-4000-b000-000000000000
  Characteristic: f000aa02-0451-4000-b000-000000000000
  Characteristic: f000aa03-0451-4000-b000-000000000000
```

The same is true when reading characteristic values:
```dart
peripheral.readCharacteristic(service.uuid.toUpperCase(), characteristic.uuid);
```
which throws the exception:
```json
SimulatedBleError({"errorCode":404,"reason":"Characteristic f000aa01-0451-4000-b000-000000000000 not found"})
```

Tested on Android Emulator API29.

**Comment by mikolak** [comment]
Gah, thought this was fixed. Could you see if this occurs on iOS as well? `.services()` and `.characteristics()` work from a local cache in the native code instead of asking Dart for the results each time (quirk caused by the interface definition of BleAdapter), so it might be platform specific.

I'll fix it this or next week.

**Comment by mikolak** [comment]
One more thing occurred to me: could you see if using `await service.characteristics()` results in the same issue?

**Comment by nicklars3n** [comment]
I can't test iOS unfortunately.

Using `await service.characteristics()` works for me.

It seems to just be an issue when specifying uuids with `await peripheral.characteristics(...)` and `await peripheral.readCharacteristic(...)`.

**Comment by mikolak** [comment]
Fixed 1.1.2, releasing in about 30 minutes.
